pkg/util/cluster/cluster.go

```diff
@@ -356,22 +357,103 @@ func (c *Cluster) Create(ctx context.Context, vnetResourceGroup, clusterName str
 	return nil
 }
 
+// ipRangesContainCIDR checks, weather any of the ipRanges overlap with the cidr string. In case cidr isn't valid, false is returned.
+func ipRangesContainCIDR(ipRanges []*net.IPNet, cidr string) bool {
+	_, cidrNet, err := net.ParseCIDR(cidr)
+	if err != nil {
+		return false
+	}
+
+	for _, snet := range ipRanges {
+		if snet.Contains(cidrNet.IP) || cidrNet.Contains(snet.IP) {
+			return true
+		}
+	}
+	return false
+}
+
+// GetIpRangesFromSubnet converts a given azure subnet to a list if IPNets.
+// Because an az subnet can cover multiple ipranges, we need to return a slice
+// instead of just a single ip range. This function never errors. If something
+// goes wrong, it instead returns an empty list.
+func GetIpRangesFromSubnet(subnet mgmtnetwork.Subnet) []*net.IPNet {
+	ipRanges := []*net.IPNet{}
+	if subnet.AddressPrefix != nil {
+		_, ipRange, err := net.ParseCIDR(*subnet.AddressPrefix)
+		if err == nil {
+			ipRanges = append(ipRanges, ipRange)
+		}
+	}
+
+	if subnet.AddressPrefixes == nil {
+		return ipRanges
+	}
+
+	for _, snetPrefix := range *subnet.AddressPrefixes {
+		_, ipRange, err := net.ParseCIDR(snetPrefix)
+		if err == nil {
+			ipRanges = append(ipRanges, ipRange)
+		}
+	}
+	return ipRanges
+}
+
+
+// getAllDevSubnets queries azure to retrieve all subnets assigned the vnet
+// `dev-vnet` in the current resource group
+func (c *Cluster) getAllDevSubnets() ([]mgmtnetwork.Subnet, error) {
+	allSubnets := []mgmtnetwork.Subnet{}
+	availSnetResults, err := c.subnets.List(context.TODO(), c.env.ResourceGroup(), "dev-vnet")
+	if err != nil {
+		return allSubnets, err
+	}
+	allSubnets = append(allSubnets, availSnetResults.Values()...)
+	for availSnetResults.NotDone() {
+		err = availSnetResults.NextWithContext(context.TODO())
```

Comment:
Super small nit: I think context.TODO() is usually used to indicate that we are not sure what context to use and hence we create one empty but using TODO to indicate that it probably needs to be revisited in the future. 

I think that in our case, context.Background() is a bit better as it indicates that it is a plain new context. Just a semantic difference that has no impact on the result 🙂